### PR TITLE
Init

### PR DESCRIPTION
**Pull request by rhpvorderman**
Tests and more extensive documentation (sphinx-argparse!) to be added later.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+The algorithm is a reimplementation from [biopet-fastqsplitter](
+https://github.com/biopet/fastqsplitter). Fastqsplitter reads a fastq
+file. It then splits the reads over the designated output files.
+
+fastqsplitter uses the excellent [xopen library by @marcelm](
+https://github.com/marcelm/xopen). This determines by extension whether the 
+file is compressed. If the output files for fastqsplitter end with '.gz' they 
+will be gzip compressed automatically. The default compression level is 1 and
+can be changed by using the `-c` flag.
```
Comment (sndrtj) [inline README.md:15]: Perhaps `-l | --level` would be more suited?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+|                       | fastqsplitter (defaults) | fastqsplitter -t 4 -c 5 |  biopet-fastqsplitter |
+| --------------------  | ------------------------ | ----------------------- | --------------------- |
+|real time              | 0m57.632s                | 1m30.291s               | 1m41.385s             |
+|total cpu time         | 3m33.276s                | 8m36.600s               | 8m20.304s             |
+|max mem                | 24 MB                    | 32MB                    | 400MB                 |
+|max vmem               | 82 MB                    | 1.6 GB                  | 11.0 GB               |
+|output files total size| 2290 MB                  | 2025 MB                 | 2025 MB               |
+
+The outcomes for multiple runs were fairly consistent with a max +-3 second difference between runs.
```
Comment (sndrtj) [inline README.md:61]: https://github.com/sharkdp/hyperfine could also be used for benchmarking ;-)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+The used test machine had 32 GB memory (2x16GB 2133mhz), a Intel core i7-6700
+(4 cores, 8 threads) and a Sandisk X400 500gb SSD.
+
+|                       | fastqsplitter (defaults) | fastqsplitter -t 4 -c 5 |  biopet-fastqsplitter |
+| --------------------  | ------------------------ | ----------------------- | --------------------- |
+|real time              | 0m57.632s                | 1m30.291s               | 1m41.385s             |
+|total cpu time         | 3m33.276s                | 8m36.600s               | 8m20.304s             |
+|max mem                | 24 MB                    | 32MB                    | 400MB                 |
+|max vmem               | 82 MB                    | 1.6 GB                  | 11.0 GB               |
+|output files total size| 2290 MB                  | 2025 MB                 | 2025 MB               |
```
Comment (sndrtj) [inline README.md:59]: What explains the difference in file size?

File: src/fastqsplitter/__init__.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        number_of_output_files = len(output_handles)
+        fastq_eof_reached = False
+        while not fastq_eof_reached:
+            # By using modulo we pick one file at a time and make sure all
+            # files are written to equally.
+            file_to_write = output_handles[i % number_of_output_files]
+
+            # Make sure a multiple of 4 lines is written as each record is
+            # 4 lines.
+            for j in range(group_size * 4):
```
Comment (sndrtj) [inline src/fastqsplitter/__init__.py:115]: This necessarily does _not_ support fastq files with multiline sequences (those exist). That's fine - I wouldn't support this either - but it should be documented.

**Review by sndrtj — CHANGES_REQUESTED** [review]
There is a test file, but no tests?

File: README.md (hunk 3 above)
Comment (rhpvorderman) [reply README.md:59]: As mentioned above:

> Fastqsplitter runs with 1 thread per output file and compression level 1 by default.
For fair comparison with biopet-fastqsplitter, fastqsplitter was run with 4
threads per file (xopen default) and compression level 5.

File: src/fastqsplitter/__init__.py (hunk 4 above)
Comment (rhpvorderman) [reply src/fastqsplitter/__init__.py:115]: It is mentioned in the README now.

File: README.md (hunk 2 above)
Comment (rhpvorderman) [reply README.md:61]: Thanks! I will look into that.

**Comment by rhpvorderman** [comment]
I will add tests and sphinx documentation later.

**Review by sndrtj — APPROVED** [review]

**Pull request merged**